iosVideo/iosVideo/CameraViewController.swift

```diff
@@ -0,0 +1,146 @@
+        cameraView.videoPlayerView.videoGravity = .resizeAspectFill
+        setUpCaptureSession()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        captureSession.startRunning()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        captureSession.stopRunning()
+    }
+
+    private func updateViews() {
+        recordButton.isSelected = fileOutput.isRecording
+    }
+
+    // Live Preview + inputs/outputs
+
+    private func setUpCaptureSession() {
+        // Add inputs
+        captureSession.beginConfiguration()
+        // Camera input
+        let camera = bestCamera()
+
+        guard let cameraInput = try? AVCaptureDeviceInput(device: camera),
+            captureSession.canAddInput(cameraInput) else {
+                fatalError("Error adding camera to capture session")
+        }
+        captureSession.addInput(cameraInput)
+        // Microphone input
+        if captureSession.canSetSessionPreset(.hd1920x1080) {
+            captureSession.sessionPreset = .hd1920x1080
+        }
+
+        // Add outputs
+        guard captureSession.canAddOutput(fileOutput) else {
+            fatalError("Error: Cannot save movie")
+        }
+        captureSession.addOutput(fileOutput)
+
+        captureSession.commitConfiguration()
+        cameraView.session = captureSession
+
+        // TODO: Start/Stop session
+
+    }
+
+    private func bestCamera() -> AVCaptureDevice {
+        // Ultra-wide lens (iPhone 11 Pro + Pro Max on back)
+        if let ultraWideCamera =  AVCaptureDevice.default(.builtInUltraWideCamera, for: .video, position: .back) {
+            return ultraWideCamera
+        }
+        // Wide angle lens ( available on any iPhone - front/back)
+        if let wideAngleCamera = AVCaptureDevice.default(.builtInWideAngleCamera, for: .video, position: .back) {
+            return wideAngleCamera
+        }
+        // No cameras present ( simulator )
+        fatalError("No camera available - are you on a simmulator?")
+    }
+
+    @IBAction func cameraButtonPressed(_ sender: Any) {
+        toggleRecord()
+    }
+
+    private func toggleRecord() {
+        if fileOutput.isRecording {
+            fileOutput.stopRecording()
+        } else {
+            fileOutput.startRecording(to: newRecordingURL(), recordingDelegate: self)
+        }
+    }
+
+    /// Creates a new file URL in the documents directory
+    private func newRecordingURL() -> URL {
+        let documentsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
+
+        let formatter = ISO8601DateFormatter()
+        formatter.formatOptions = [.withInternetDateTime]
+
+        let name = formatter.string(from: Date())
+        let fileURL = documentsDirectory.appendingPathComponent(name).appendingPathExtension("mov")
+        return fileURL
+    }
+
+    // MARK: - Navigation
+
+    var editURL: URL!
```

Comment:
Is there a reason this isn't at the top with the other values? I know it's used in the `prepare` method but I know personally I generally prefer to put this sort of thing at the top anyways. :D